control-plane-operator/controllers/hostedcontrolplane/csi/kubevirt/kubevirt.go

```diff
@@ -250,6 +256,40 @@ func reconcileController(controller *appsv1.Deployment, componentImages map[stri
 		}
 	}
 
+	if configoperator.IsExternalInfraKv(hcp) {
+		csiDriverContainerIndex := func() int {
+			for i, container := range controller.Spec.Template.Spec.Containers {
+				if container.Name == "csi-driver" {
+					return i
+				}
+			}
+			return -1
+		}
+		if csiDriverContainerIndex() == -1 {
+			panic(fmt.Errorf("unable to find csi-driver container in %s pod", controllerDeployment.Name))
+		}
+		csiDriverContainer := controller.Spec.Template.Spec.Containers[csiDriverContainerIndex()]
```

Comment:
let's call it once. Also, can we avoid this lambda?

```suggestion
		containerIndex := csiDriverContainerIndex()
		if containerIndex == -1 {
			panic(fmt.Errorf("unable to find csi-driver container in %s pod", controllerDeployment.Name))
		}
		csiDriverContainer := controller.Spec.Template.Spec.Containers[containerIndex()]
```